neqo-bin/src/client/mod.rs

```diff
@@ -493,6 +493,14 @@ fn qlog_new(args: &Args, hostname: &str, cid: &ConnectionId) -> Res<NeqoQlog> {
     .map_err(Error::QlogError)
 }
 
+const fn unspecified_addr(addr: &SocketAddr) -> SocketAddr {
+    match addr {
+        SocketAddr::V4(..) => SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), 0),
+        SocketAddr::V6(..) => SocketAddr::new(IpAddr::V6(Ipv6Addr::UNSPECIFIED), 0),
+    }
+}
+
+#[allow(clippy::too_many_lines)]
```

Comment:
This is usually a good sign that some work is needed.  Offhand, I see a nice candidate on line 517 (to 532).